packages/p2p/src/components/verification/verification.tsx

```diff
@@ -1,73 +1,101 @@
 import React from 'react';
-import { observer } from 'mobx-react-lite';
 import { Icon, Checklist, Text } from '@deriv/components';
-import { isDesktop, isMobile, routes } from '@deriv/shared';
+import { routes } from '@deriv/shared';
+import { observer } from '@deriv/stores';
 import Dp2pBlocked from 'Components/dp2p-blocked';
 import { Localize } from 'Components/i18next';
-import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
-import NicknameForm from 'Components/nickname-form';
+import { document_status_codes, identity_status_codes } from 'Constants/account-status-codes';
 import { useStores } from 'Stores/index';
 
 const Verification = () => {
     const { general_store } = useStores();
-    const { showModal } = useModalManagerContext();
-
-    if (!general_store.is_advertiser && general_store.poi_status === 'verified' && general_store.nickname) {
-        return <Dp2pBlocked />;
-    }
-
-    if (isMobile() && general_store.should_show_popup) {
-        return <NicknameForm />;
-    }
-
+    const { p2p_poa_required, poa_status, poi_status } = general_store;
+    const getPoiAction = (status: string) => {
+        switch (status) {
+            case identity_status_codes.PENDING:
+                return <Localize i18n_default_text='Your proof of identity is still under review.' />;
+            case identity_status_codes.REJECTED:
+                return <Localize i18n_default_text='Identity verification failed. Please try again.' />;
+            case identity_status_codes.VERIFIED:
+                return <Localize i18n_default_text='Your proof of identity is verified.' />;
+            default:
+                return <Localize i18n_default_text='Upload your documents to verify your identity.' />;
+        }
+    };
+    const getPoaAction = (status: string) => {
+        switch (status) {
+            case document_status_codes.PENDING:
+                return <Localize i18n_default_text='Your proof of address is still under review.' />;
+            case document_status_codes.REJECTED:
+                return <Localize i18n_default_text='Address verification failed. Please try again.' />;
+            case document_status_codes.VERIFIED:
+                return <Localize i18n_default_text='Your proof of address is verified.' />;
+            default:
+                return <Localize i18n_default_text='Upload your documents to verify your address.' />;
+        }
+    };
```

Comment:
since these two functions dont depend on any thing inside of the component, could we move it out of it? or we could wrap them with useMemo to avoid any unnecessary re-renders to these functions